### PR TITLE
Make the 'PSISERemoteSessionOpenFile' a support event

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Make the `PSISERemoteSessionOpenFile` a support event, so `Get-EventSubscriber` won't show up that subscriber. `Get-EventSubscriber -Force` can still show the support events. `Unregister-Event -Force` needs to be used to remove a support event.

The event subscriber for `PSInternalRemoteDebuggerStopEvent` and `PSInternalRemoteDebuggerBreakpointUpdatedEvent` are already made support events, see [`ServerRunspacePoolDriver.cs`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/remoting/server/ServerRunspacePoolDriver.cs#L2167).

And this PR makes it the same for `PSISERemoteSessionOpenFile`.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@PaulHigin and @adityapatwardhan, can you please take a look?

**Review by PaulHigin — APPROVED** [review]

**Review by PaulHigin — CHANGES_REQUESTED** [review]
Whoops, I meant to leave comment with requested change (code clean up should include method name and comment changes).

**Review by PaulHigin — APPROVED** [review]
I changed my mind yet again.  We can't change the SourceIdentifier name without breaking VSCode.  So I feel these changes are good, just please ensure VSCode remote debug/editing still works.

**Review by adityapatwardhan — APPROVED** [review]

File: src/System.Management.Automation/engine/hostifaces/HostUtilities.cs [hunk 1]
```diff
@@ -943,7 +943,7 @@ public static Collection<PSObject> InvokeOnRunspace(PSCommand command, Runspace
                 [string] $PSEditFunction
             )
 
-            Register-EngineEvent -SourceIdentifier PSISERemoteSessionOpenFile -Forward
+            Register-EngineEvent -SourceIdentifier PSISERemoteSessionOpenFile -Forward -SupportEvent
```
[the remark this answers is not in the document]
Comment (daxian-dbw) [reply src/System.Management.Automation/engine/hostifaces/HostUtilities.cs]: This is just code cleanup. I found the forwarding events `PSInternalRemoteDebuggerStopEvent` and `PSInternalRemoteDebuggerBreakpointUpdatedEvent` are both supporting events. `Get-EventSubscriber` doesn't return them but return `PSISERemoteSessionOpenFile`, so I submitted this PR to make them consistent.

I will sync with Tyler to verify `PSEdit` with VS code.

**Comment by daxian-dbw** [comment]
It turns out VS Code uses a different event: `PSESRemoteSessionOpenFile`, so this change won't affect VS Code.

But they don't use `-SupportEvent` too, so I will submit a PR to `PowerShellEditorServices` repo to fix that.

**Pull request merged**